src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from "react";
+import "twin.macro";
+import "styled-components/macro";
+import {
+  Select,
+  Button,
+  AutoComplete,
+} from "@madie/madie-design-system/dist/react";
+import ExpandingSection from "../../../common/ExpandingSection";
+import { MenuItem } from "@mui/material";
+import { getNameOptionsByType } from "./ExpressionEditorHelper";
+import { ControlledTextarea } from "../../../common/ControlledTextArea";
+
+interface ExpressionsProps {
+  canEdit: boolean;
+  expressionEditorOpen: boolean;
+  formik: any;
+  expressionValue: string;
+  setExpressionValue: Function;
+}
+
+export default function ExpressionEditor(props: ExpressionsProps) {
+  const {
+    canEdit,
+    expressionEditorOpen,
+    formik,
+    expressionValue,
+    setExpressionValue,
+  } = props;
+  const [namesOptions, setNamesOptions] = useState([]);
+
+  const availableTypes = [
+    "Parameters",
+    "Definitions",
+    "Functions",
+    "Timing",
+    "Pre-Defined Functions",
+  ];
+
+  const renderMenuItems = (options: string[]) => {
+    return [
+      ...options.map((value) => (
+        <MenuItem
+          key={`${value}-option`}
+          value={value}
+          data-testid={`${value}-option`}
+        >
+          {value}
+        </MenuItem>
+      )),
+    ];
+  };
+
+  return (
+    <div>
+      <ExpandingSection
+        title="Expression Editor"
+        showHeaderContent={expressionEditorOpen}
+        children={
+          <>
+            <div tw="flex flex-wrap">
+              <div tw="w-1/2">
+                <Select
+                  label="Type"
+                  id={"type-selector"}
+                  inputProps={{
+                    "data-testid": "type-selector-input",
+                  }}
+                  data-testid={"type-selector"}
+                  SelectDisplayProps={{
+                    "aria-required": "true",
+                  }}
+                  renderValue={(val) => {
+                    if (val) {
+                      const type = availableTypes.find((el) => el === val);
+                      setNamesOptions(getNameOptionsByType(type));
+                      return type;
+                    }
+                    return val;
+                  }}
+                  options={renderMenuItems(availableTypes)}
+                  disabled={!canEdit}
+                  {...formik.getFieldProps("type")}
```

Comment:
I think ideally this should come first before all other overridden props

Reply:
@adongare I just made the change, can you please check again? Thanks so much!!